pennylane/data/data_manager/graphql.py

```diff
@@ -0,0 +1,155 @@
+"""
+Module for containing graphql functionality for interacting with the Datasets Service API.
+"""
+
+import os
+from typing import Any, Optional
+
+from requests import post
+
+GRAPHQL_URL = os.getenv("DATASETS_ENDPOINT_URL", "https://cloud.pennylane.ai/graphql")
+
+
+class GraphQLError(BaseException):
+    """Exception for GraphQL"""
+
+
+def _get_graphql(url: str, query: str, variables: Optional[dict[str, Any]] = None):
```

Comment:
Since these functions are used outside the module they shouldn't be underscored
```suggestion
def get_graphql(url: str, query: str, variables: Optional[dict[str, Any]] = None):
```